Add more kinetics!

kdorheim opened the pull request:
Add POMdecomp & MBdecay kinetics.

kdorheim commented on R/aadoc.R:
```diff
@@ -1,2 +1,19 @@
+#' \item{RMM}{Reverse Michaelis-Menten kinetics, TODO add ref}
+#' \item{ECA}{Equilibrium chemistry approximation, based on RESOM (Tang and Riley, 2014)}
+#' \item{LM}{Linear kinetics model, TODO add ref}
+#' }
```
@jianqiuz do we need to add references for the Michaelis–Menten kinetics, Reverse Michaelis-Menten kinetics ad, linear kinetics model?

jianqiuz replied:
RMM ref: Schimel, J. P., and M. N. Weintraub (2003), The implications of exoenzyme activity on microbial carbon and nitrogen limitation in soil: A theoretical model, Soil Biol. Biochem., 35(4), 549–563, doi:10.1016/s0038-0717(03)00015-4; Sulman, B. N., R. P. Phillips, A. C. Oishi, E. Shevliakova, and S. W. Pacala (2014), Microbe-driven turnover offsets mineral-mediated storage of soil carbon under elevated CO2, Nat. Clim. Change, 4(12), 1099–1102, doi:10.1038/nclimate2436.

Linear ref: Robertson, A. D., Paustian, K., Ogle, S., Wallenstein, M. D., Lugato, E., and Cotrufo, M. F.: Unifying soil organic matter formation and persistence frameworks: the MEMS model, Biogeosciences, 16, 1225–1248, https://doi.org/10.5194/bg-16-1225-2019, 2019.

kdorheim replied:
Awesome thanks!

kdorheim commented on R/data.R:
We have corpse, commission, and mend, do we want to add any other model configurations?

jianqiuz replied:
No additional configurations at this point. Will need to enable toy model---user-defined configuration

kdorheim commented on R/model_componets.R:
```diff
@@ -174,68 +177,118 @@ carbon_fluxes_internal <- function(env, state = NULL, params = NULL){
+        } else if (DOMdecomp == "LM") {
+          fluxes[["F1"]] = function() {
+            # DOC uptake by microbial biomass, note that this uses a linear model of kinetics.
+            V.d * B * D / (K.d + B)
```
Is this correct for the linear model? or should it look more along the lines of 

`(1 / E.c) * (V.d + m.r) * B * D / (K.d + B)` ?

kdorheim commented on R/model_componets.R:
@jianqiuz right now we only have the LM of MB decay implemented, is there another form we should add?

jianqiuz replied:
The alternative kinetic for MBdecay is not MM, it is Density-dependent microbial turnover (Georgiou paper). So we only have two options for MBdecay, Linear or DD

kdorheim replied:
Is https://www.nature.com/articles/s41467-017-01116-z the Georgiou paper you are referring to? 

For the DD would it look like? 

`1 - p.ep - p.em) * m.r * B * (1 - B/K)` where K is the carrying capacity? Is there a default value we would want to use?

jianqiuz replied:
The formulation v * B * (1 - B/K) represents the logistic growth (with carrying capacity of the population). It can be rearranged to v*B-v*B^2/K, where v*B is the growth rate, which is our F1, V.d * B * D / (K.d + B). The second term        -v*B^2/K represents constrained microbial biomass, which is further rearranged to -k.B*B^2. This formulation of biomass decay follows the same form as our linear decay model: -k*B. In this way, we're introducing a new parameter "beta", the decay=-k*B^(beta). The default value of beta is 1, which gives us the linear model, while beta >1 represents density-dependent biomass decay. 

So for both the linear and DD microbial biomass decay, it looks like (1 - p.ep - p.em) * m.r * B ^beta. In linear formation, beta=1

kdorheim commented:
@jianqiuz please take a look at the density-dependent MB decay that I added, [here](https://rpubs.com/kdorheim/869346 ) is an example walking through it, this example will be included with the package as a vignette and in the package manual after this PR is merged. If you think it is useful I wonder if we would want to add more text to this manuscript talking about the different kinetics options we have for the other fluxes. 

What additional information would be useful to add there? Should the name of the example be changed to something else like kinetics & dynamics? or microbial explicit dynamics?

I have also added some notes to the READ ME about how to update the package to capture new developments/changes on git.
